app/ui/browser/model/index.js

```diff
@@ -66,6 +66,9 @@ export const UIState = Immutable.Record({
 
   // What the user has typed into the location bar, stored by page id
   userTypedLocation: Immutable.Map(),
+
+  // Should the autocomplete popup be visible
+  showCompletions: true,
```

Comment:
We have some logic in the local state for the locations component right now that hides the popup if the url bar isn't visible / focused, which would be the case when switching tabs.  I think we should move these bits of state up to the UI state using the same pattern as here.  Not sure that indexing completion by page id is necessary right now although it might become eventually.